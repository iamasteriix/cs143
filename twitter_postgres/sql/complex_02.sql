/*
 * Like the query in 01.sql that calculates the hashtags that are commonly used with the hashtag #coronavirus,
 * but also returns the total number of times each hashtag is used.
 */
SELECT t1.tag, shared_count, total_count
FROM (
    SELECT lower(t1.tag) as tag,count(*) as shared_count
    FROM tweet_tags t1
    INNER JOIN tweet_tags t2 ON t1.id_tweets = t2.id_tweets
    WHERE
        lower(t2.tag)='#coronavirus'
    GROUP BY (1)
    LIMIT 100
) t1
INNER JOIN (
    SELECT 
        lower(tag) as tag,
        count(*) as total_count
        FROM tweet_tags
        GROUP BY (1)
    ) t3 ON t1.tag = t3.tag
ORDER BY shared_count DESC,tag
;
