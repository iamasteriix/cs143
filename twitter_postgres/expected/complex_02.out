           tag            | shared_count | total_count 
--------------------------+--------------+-------------
 #coronavirus             |           22 |          22
 #covid19                 |            4 |          20
 #1u                      |            1 |           1
 #2a                      |            1 |           1
 #aprilfoolsday           |            1 |           7
 #californialockdown      |            1 |           2
 #china                   |            1 |           1
 #clapfordchealthworkers  |            1 |           1
 #coronavirusdeutschland  |            1 |           1
 #coronavirusnewyork      |            1 |           1
 #covidー19               |            1 |           9
 #covid_19gr              |            1 |           2
 #covıd19                 |            1 |           3
 #cuarentena              |            1 |           5
 #dc                      |            1 |           1
 #deutschland             |            1 |           1
 #dog                     |            1 |           1
 #escapedog               |            1 |           1
 #government              |            1 |           1
 #hermes                  |            1 |           1
 #iambatman               |            1 |           1
 #instadog                |            1 |           1
 #losangeleslockdown      |            1 |           1
 #musik                   |            1 |           1
 #perro                   |            1 |           1
 #prisionbreak            |            1 |           1
 #quarantine              |            1 |           7
 #quarentena              |            1 |           2
 #quedateencasa           |            1 |          14
 #rechauffementclimatique |            1 |           1
 #santaclarita            |            1 |           1
 #statenisland            |            1 |           1
 #stayathome              |            1 |          15
 #stayhome                |            1 |           5
 #thefinal5               |            1 |           1
 #tyranny                 |            1 |           1
 #uklockdown              |            1 |           1
 #unnouveausujetparjour   |            1 |           1
 #yeahthatgreenville      |            1 |           1
(39 rows)

